src/Two/LinkedInProvider.php

```diff
@@ -24,99 +24,64 @@ class LinkedInProvider extends AbstractProvider implements ProviderInterface
-            RequestOptions::HEADERS => [
-                'Authorization' => 'Bearer '.$token,
-                'X-RestLi-Protocol-Version' => '2.0.0',
-            ],
-            RequestOptions::QUERY => [
-                'projection' => '(id,firstName,lastName,profilePicture(displayImage~:playableStreams))',
-            ],
-        ]);
-
-        return (array) json_decode($response->getBody(), true);
-    }
-
-    /**
-     * Get the email address for the user.
      *
-     * @param  string  $token
-     * @return array
+     * @throws GuzzleException
      */
-    protected function getEmailAddress($token)
+    protected function getBasicProfile($token): array
     {
-        $response = $this->getHttpClient()->get('https://api.linkedin.com/v2/emailAddress', [
+        $response = $this->getHttpClient()->get('https://api.linkedin.com/v2/userinfo', [
             RequestOptions::HEADERS => [
                 'Authorization' => 'Bearer '.$token,
                 'X-RestLi-Protocol-Version' => '2.0.0',
             ],
             RequestOptions::QUERY => [
-                'q' => 'members',
-                'projection' => '(elements*(handle~))',
+                'projection' => '(sub,email,given_name,family_name,picture)',
             ],
         ]);
 
-        return (array) Arr::get((array) json_decode($response->getBody(), true), 'elements.0.handle~');
+        return (array) json_decode($response->getBody(), true);
     }
 
     /**
      * {@inheritdoc}
      */
-    protected function mapUserToObject(array $user)
+    protected function mapUserToObject(array $user): User
     {
-        $preferredLocale = Arr::get($user, 'firstName.preferredLocale.language').'_'.Arr::get($user, 'firstName.preferredLocale.country');
-        $firstName = Arr::get($user, 'firstName.localized.'.$preferredLocale);
-        $lastName = Arr::get($user, 'lastName.localized.'.$preferredLocale);
-
-        $images = (array) Arr::get($user, 'profilePicture.displayImage~.elements', []);
-        $avatar = Arr::first($images, function ($image) {
-            return $image['data']['com.linkedin.digitalmedia.mediaartifact.StillImage']['storageSize']['width'] === 100;
-        });
-        $originalAvatar = Arr::first($images, function ($image) {
-            return $image['data']['com.linkedin.digitalmedia.mediaartifact.StillImage']['storageSize']['width'] === 800;
-        });
-
         return (new User)->setRaw($user)->map([
-            'id' => $user['id'],
+            'id' => $user['sub'],
             'nickname' => null,
-            'name' => $firstName.' '.$lastName,
-            'first_name' => $firstName,
-            'last_name' => $lastName,
-            'email' => Arr::get($user, 'emailAddress'),
-            'avatar' => Arr::get($avatar, 'identifiers.0.identifier'),
-            'avatar_original' => Arr::get($originalAvatar, 'identifiers.0.identifier'),
+            'name' => $user['name'].' '.$user['family_name'],
```

Comment:
in this part:

`'name' => $user['name'].' '.$user['family_name'],`

Should it be `$user['name']` or `$user['given_name']?

Reply:
Indeed, it should be $user['given_name']. Thanks